SgmlTests/Tests-Logic.cs

```diff
@@ -82,19 +82,24 @@ private enum XmlRender {
         private static void ReadTest(string name, out string before, out string after) {
             var assembly = typeof(Tests).Assembly;
            
+            var test = ReadTestResource(name).Split('`');
+            before = test[0];
+            after = test[1];
+        }
+
+        internal static string ReadTestResource(string name) {
```

Comment:
The changes here from `private` to `internal` and minor refactoring was also related to my `LoggingXmlReader` change that I didn't fully undo.

Reply:
this is a good refactoring thanks.  I think LoadDtd could also use your new ReadTestResource, although it would have to switch to a StringReader...

Reply:
I did change the test resource-loading code to use `LoadDtd` already, did you overlook it? or did you mean something else?

Reply:
Ah missed it then we can resolve this conversation, you finished with the PR then?

Reply:
After this PR is merged is it okay for me to resubmit the other PR concerned with splitting-up the files?